### PR TITLE
Copy metadata to avoid race condition in forwarding

### PR DESCRIPTION
**Pull request by lftakakura**

### CODE REVIEW

**Comment by andrehp** [comment]
In what scenario did this race condition happen?

File: plugins/grpc/forwarder.go [hunk 1]
```diff
@@ -349,7 +349,13 @@ func (g *GRPCForwarder) Forward(ctx context.Context, event string, infos, fwdMet
 
 	// Add forwarder metadata (from maestro config) to request
 	// Client metadata should take priority
-	metadata := g.metadata
+	metadata := map[string]interface{}{}
+	if g.metadata != nil {
+		for k, v := range g.metadata {
```
Comment (ghostec) [inline plugins/grpc/forwarder.go]: do you see any problem in doing a shallow copy?

**Comment by lftakakura** [comment]
> In what scenario did this race condition happen?

Before `metadata` was pointing to `g.metadata`. As `g.metadata` is global for the app, the service was crashing frequently as it was trying to read and write during forwardings

**Comment by cscatolini** [comment]
> > In what scenario did this race condition happen?
> 
> `metadata` was pointing to `g.metadata`. As `g.metadata` is global for the app, the service was crashing frequently as it was trying to read and write during forwardings

It makes no sense for metadata to be an attribute of the struct 🤔 

This change LGTM.

**Pull request merged**